recipes/cgns/all/conanfile.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class CgnsConan(ConanFile):
+    name = "cgns"
+    description = "Standard for data associated with the numerical solution " \
+                  "of fluid dynamics equations."
+    topics = ("conan", "cgns", "data", "cfd", "fluids")
+    homepage = "http://cgns.org/"
+    license = "zlib/libpng"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_hdf5": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_hdf5": True
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_hdf5:
+            self.requires("hdf5/1.12.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("CGNS-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["CGNS_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["CGNS_ENABLE_HDF5"] = self.options.with_hdf5
+        self._cmake.configure()
+
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("license.txt", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "CGNS"
+        self.cpp_info.names["cmake_find_package_multi"] = "CGNS"
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
https://github.com/CGNS/CGNS/blob/7cc605021cc6c278acf2e69c5c3bd69ff5ee504e/src/CMakeLists.txt#L648-L654

Since both as installed it can be messy with link order... you need to be explicit with the libs

```suggestion
        target = "cgns_shared" if self.options.shared else "cgns_static"
        self.cpp_info.components[target].libs = [target]
```

Reply:
First of all: thank you for all the input! I appreciate it :D

When trying to introduce the components cgns_static and cgns_shared, I noticed the following:
The CGNS cmake files modify the output names of the target to mismatch the name of the target
(https://github.com/CGNS/CGNS/blob/7cc605021cc6c278acf2e69c5c3bd69ff5ee504e/src/CMakeLists.txt#L624-L636)
In the end the lib name is: libcgns.a, libcgns.so ... 
This means that when specifying the libs in the package info, I need to be explicit with the file ending.

Do you still prefer the way you suggested, or the original way that modifies the cmake file with a patch to obtain either the static or dynamic library?

Reply:
Thank you for putting in the work to add this very difficult project! ❤️ 

I do not like either optional 🙃 given the complexity of using the components it would be very acceptable to go with the patch.

If you go with the patch, please note these comments in the package_info so we do not lose the information you found

Reply:
Alright, then I will go for the patch.
I added a comment that explains the difference in target and output name.
Furthermore I am now explicit about the library name instead of using tools.collect_libs().
I think this helps to understand the recipe.